lambdas/pkgselect/index.py

```diff
@@ -90,172 +86,227 @@ def file_list_to_folder(df: pd.DataFrame, limit: int, offset: int) -> dict:
         objects = []
         total_results = 0
 
-    returned_results = len(prefixes) + len(objects)
     return dict(
         total=total_results,
-        returned=returned_results,
         prefixes=prefixes,
-        objects=objects
+        objects=objects,
     )
 
 
-def create_s3_client(
-    *,
-    aws_access_key_id: str,
-    aws_secret_access_key: str,
-    aws_session_token: str
-):
+def object_is_publicly_readable(bucket: str, key: str):
+    try:
+        boto3.client(
+            "s3",
+            config=botocore.client.Config(signature_version=botocore.UNSIGNED),
+        ).head_object(Bucket=bucket, Key=key)
```

Comment:
I'd have a separate [`lru_cache`](https://docs.python.org/3/library/functools.html#functools.lru_cache)d function for creation of client, because at least locally it's quite expensive (2 ms).

Reply:
it's only called once, so we won't benefit from caching it (well, maybe in localmode case only)

Reply:
Cache will be used when lambda's environment is reused between invocations, you can read some details at the end of this [page](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-context.html).